ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py

```diff
@@ -635,3 +635,37 @@ def test_tensorflow_dropout(
         prob=prob,
         scale=scale,
     )
+
+
+@handle_cmd_line_args
+@given(
+    x_f_d_df=_x_and_filters(
+        dtypes=helpers.get_dtypes("float", full=False),
+        data_format=st.sampled_from(["NHWC"]),
+        padding=st.sampled_from(["VALID", "SAME"]),
+        type="2d",
+    ),
+    ksize=helpers.array_values(dtype=ivy.float16, shape=(3, 5)),
+    num_positional_args=helpers.num_positional_args(
+        fn_name="ivy.functional.frontends.tensorflow.max_pool2d"
+    ),
+    native_array=helpers.list_of_length(x=st.booleans(), length=2),
```

Comment:
Hey @yis11178 you are generating data for convolution. I suggest you use `arrays_for_pooling` in helpers. Use this test function for your reference [here](https://github.com/unifyai/ivy/blob/master/ivy_tests/test_ivy/test_functional/test_extensions.py#L321).